rasa/shared/core/domain.py

```diff
@@ -1116,14 +1132,23 @@ def get_active_states(self, tracker: "DialogueStateTracker") -> State:
         return self._clean_state(state)
 
     def states_for_tracker_history(
-        self, tracker: "DialogueStateTracker"
+        self, tracker: "DialogueStateTracker", omit_unset_slots: bool = False,
     ) -> List[State]:
-        """Array of states for each state of the trackers history."""
+        """List of states for each state of the trackers history.
+
+        Args:
+            tracker: dialog state tracker containing the dialog so far
+            omit_unset_slots: If `True` do not include the initial values of slots.
+
+        Returns A `State` for each prior tracker.
+        """
         return [
-            self.get_active_states(tr) for tr in tracker.generate_all_prior_trackers()
+            self.get_active_states(tr, omit_unset_slots=omit_unset_slots)
+            for tr in tracker.generate_all_prior_trackers()
         ]
 
     def slots_for_entities(self, entities: List[Dict[Text, Any]]) -> List[SlotSet]:
+        """Returns a list of `SlotSet` events from a list of entities."""
```

Comment:
optional: I'd try to avoid duplicating specific types in the docstrings where possible 
```suggestion
        """Returns `SlotSet` events for extracted entities."""
```